File: app/src/main/java/br/com/jonathanzanella/myexpenses/helpers/ResourcesHelper.java
package br.com.jonathanzanella.myexpenses.helpers;

import android.content.Context;
import android.support.annotation.StringRes;

import java.lang.ref.WeakReference;

public class ResourcesHelper {
	private WeakReference<Context> contextWeakReference;

	public ResourcesHelper(Context context) {
		this.contextWeakReference = new WeakReference<>(context);
	}

	public String getString(@StringRes int string) {
		return contextWeakReference.get().getString(string);
	}
}

File: app/src/main/kotlin/br/com/jonathanzanella/myexpenses/helpers/ResourcesHelper.kt
package br.com.jonathanzanella.myexpenses.helpers

import android.content.Context
import android.support.annotation.StringRes

import java.lang.ref.WeakReference

open class ResourcesHelper(context: Context) {
    private val contextWeakReference: WeakReference<Context> = WeakReference(context)

    fun getString(@StringRes string: Int): String = contextWeakReference.get()!!.getString(string)
}
